clap_lex/src/lib.rs

```diff
@@ -408,8 +412,8 @@ impl<'s> ShortFlags<'s> {
     /// Does the short flag look like a number
     ///
     /// Ideally call this before doing any iterator
-    pub fn is_number(&self) -> bool {
-        self.invalid_suffix.is_none() && self.utf8_prefix.as_str().parse::<f64>().is_ok()
+    pub fn looks_like_number(&self) -> bool {
+        self.invalid_suffix.is_none() && looks_like_number(self.utf8_prefix.as_str())
```

Comment:
I think this can be named `looks_like_negative_number`, we just know it starts with `-` already so we don't need to check that.